test/e2e/multicast.go

```diff
@@ -0,0 +1,130 @@
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+const (
+	mcastSource  = "pod-client"
+	mcastServer1 = "pod-server1"
+	mcastServer2 = "pod-server2"
+)
+
+var _ = ginkgo.Describe("Multicast", func() {
+
+	fr := framework.NewDefaultFramework("multicast")
+
+	type nodeInfo struct {
+		name   string
+		nodeIP string
+	}
+
+	var (
+		cs                             clientset.Interface
+		ns                             string
+		clientNodeInfo, serverNodeInfo nodeInfo
+	)
+
+	ginkgo.BeforeEach(func() {
+		cs = fr.ClientSet
+		ns = fr.Namespace.Name
+
+		nodes, err := e2enode.GetBoundedReadySchedulableNodes(cs, 2)
+		framework.ExpectNoError(err)
+		if len(nodes.Items) < 2 {
+			framework.Skipf(
+				"Test requires >= 2 Ready nodes, but there are only %v nodes",
+				len(nodes.Items))
+		}
+
+		ips := e2enode.CollectAddresses(nodes, v1.NodeInternalIP)
+
+		clientNodeInfo = nodeInfo{
+			name:   nodes.Items[0].Name,
+			nodeIP: ips[0],
+		}
+
+		serverNodeInfo = nodeInfo{
+			name:   nodes.Items[1].Name,
+			nodeIP: ips[1],
+		}
+
+		// annotate namespace to enable multicast
+		namespace, err := fr.ClientSet.CoreV1().Namespaces().Get(ns, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Error getting Namespace %v: %v", fr.Namespace.Name, err)
+		if namespace.ObjectMeta.Annotations == nil {
+			namespace.ObjectMeta.Annotations = map[string]string{}
+		}
+		namespace.ObjectMeta.Annotations["k8s.ovn.org/multicast-enabled"] = "true"
+		_, err = fr.ClientSet.CoreV1().Namespaces().Update(namespace)
+		framework.ExpectNoError(err, "Error updating Namespace %v: %v", ns, err)
+	})
+
+	ginkgo.It("should be able to send multicast UDP traffic between nodes", func() {
```

Comment:
thanks @aojea and @dceara for adding these! Is there any ETA on when we could get ipv6 tests enabled? Do we need to have a github issue so we dont forget to add them later?

Reply:
It depends on the the upstream `agnhost` image update, but there are some issues with the process.
I think that the new image should be ready for next week.

Reply:
OK created: https://github.com/ovn-org/ovn-kubernetes/issues/1949